### PR TITLE
bug fix for convolutional component init()

### LINKED ISSUE #270: nnet3: Convolution1dComponent does not initialize all class members when using file based initialization.
class members like patch_dim_ are not being initialized when using "matrix" argument in the config.

### PR DESCRIPTION
**Pull request by naxingyu**
This is to resolve #270

### CODE REVIEW

**Comment by danpovey** [comment]
I'll wait till the checks finish before merging.  @vijayaditya, could you please see if the same issue exists in nnet3 and if so, ask @freewym to fix it?

**Comment by naxingyu** [comment]
I have also updated the nnet3 code. Same issue. Was a mistake.

**Comment by danpovey** [comment]
oh, I see.  Thanks.

**Comment by vijayaditya** [comment]
Solves the issue.

File: src/nnet3/nnet-simple-component.cc [hunk 1]
```diff
@@ -2082,9 +2086,12 @@ void Convolutional1dComponent::InitFromConfig(ConfigLine *cfl) {
   int32 input_dim = -1, output_dim = -1;
   int32 patch_dim = -1, patch_step = -1, patch_stride = -1;
   cfl->GetValue("learning-rate", &learning_rate); //optional
+  cfl->GetValue("patch-dim", &patch_dim);
```
Comment (vijayaditya) [inline src/nnet3/nnet-simple-component.cc]: Please modify to 
<pre><code>
ok = ok && cfl->GetValue("patch-dim", &patch_dim);
ok = ok && cfl->GetValue("patch-step", &patch_step);
ok = ok && cfl->GetValue("patch-stride", &patch_stride);
</code></pre>

the method has to ensure that the specified values exist.

Comment (naxingyu) [inline src/nnet3/nnet-simple-component.cc]: Done.

Comment (naxingyu) [inline src/nnet3/nnet-simple-component.cc]: @vijayaditya <https://github.com/vijayaditya> would you mind check the 
CI log? banned in China mainland....

**Comment by naxingyu** [comment]
I hacked around and saw the build log... anyway
It seems that the build failed at configuring openfst. Is it possible to 
rerun the build?
The code compiles correctly on my local station.

**Comment by danpovey** [comment]
@kkm000, did you change anything about the Travis build?
It can't find gcc-4.9.
